relayer/pkg/chainlink/ocr2/utils.go

```diff
@@ -217,11 +218,10 @@ func parseConfigEventData(eventData []*caigotypes.Felt) (types.ContractConfig, e
 		return types.ContractConfig{}, errors.Wrapf(err, "expected onchainConfig version 1, got %v", onchainConfigVersion)
 	}
 
-	temp := median.OnchainConfig{
+	onchainConfig, err := medianreport.OnchainConfigCodec{}.Encode(median.OnchainConfig{
```

Comment:
In this file we have another ad-hoc encoding, this time from `[]*caigotypes.Felt` to `[]byte`:

```
onchainConfigFelts := eventData[index:(index + int(onchainConfigLen))]

onchainConfigVersion := onchainConfigFelts[0].Uint64()
if onchainConfigVersion != 1 {
	return types.ContractConfig{}, errors.Wrapf(err, "expected onchainConfig version 1, got %v", onchainConfigVersion)
}

onchainConfig, err := medianreport.OnchainConfigCodec{}.Encode(median.OnchainConfig{
	Min: signedFelt(onchainConfigFelts[1]),
	Max: signedFelt(onchainConfigFelts[2]),
})
```

Could we simplify this by going directly from `[]*caigotypes.Felt` to `[]byte`?

Or we could move this logic as a helper function to `medianreport/onchainconfig.go` so we contain all these onchainconfig encoding/decoding in this one file. (notice that the version magic number is also duplicated here)